File: apps/backend/app/modules/clients/manifest.py
from __future__ import annotations

from app.modules.registry.models import ModuleManifest, Route, UiMenu

MANIFEST = ModuleManifest(
    id="clientes",
    name="Clientes",
    version="1.0.0",
    scopes=["tenant"],
    permissions=["clientes.read", "clientes.write"],
    plan_flags=["basic", "pro"],
    routes=[
        Route(path="/api/v1/clients", method="GET", scope="tenant"),
        Route(path="/api/v1/clients", method="POST", scope="tenant"),
    ],
    ui=UiMenu(title="Clientes", icon="users", route="/clients", order=20),
)

File: apps/tenant/src/modules/customers/manifest.ts
import React from 'react'

const ClientesList = React.lazy(() => import('./List'))
const ClientesForm = React.lazy(() => import('./Form'))

export const manifest = {
  id: 'customers',
  name: 'Customers',
  version: '1.0.0',
  permissions: ['customers.read', 'customers.write'],
  routes: [
    { path: '/clients', element: ClientesList },
    { path: '/clients/new', element: ClientesForm },
    { path: '/clients/:id/edit', element: ClientesForm }
  ],
  menu: {
    title: 'Customers',
    icon: '👥',
    route: '/clients',
    order: 35
  }
}
